more_itertools/more.py

```diff
@@ -770,16 +774,31 @@ def side_effect(func, iterable, chunk_size=None):
         >>> list(pair_sums)
         [1, 5, 9]
 
+    Writing to a file-like object:
+
+        >>> from io import StringIO
+        >>> from more_itertools import consume
+        >>> f = StringIO()
+        >>> func = lambda x: print(x, file=f)
+        >>> it = [u'a', u'b', u'c']
+        >>> consume(side_effect(func, it, file_obj=f))
```

Comment:
But as a kwarg, it doesn't collide with names: it's more akin to a dictionary key.